### PR TITLE
Added API error codes and implemented 404 on file endpoints

### PR DESCRIPTION
**Pull request by ghost**
## Context

Our endpoints should provide meaningful errors. This PR adds 404, and implements it on relevant file endpoints.

On the changes to the E2E tests, the existing `make test-integration` wasn't working, and the existing solution would never extend to E2E tests involving the LLM, as we can't put API keys in `.env.integration`. A blended approach of pytest using the existing `.env`, but overwriting a few key vars, makes more sense.

## Changes proposed in this pull request

* Adds pydantic error models inspired loosely by the [RFC 9457 error specification](https://www.rfc-editor.org/rfc/rfc9457.html), opening to door to following it properly in future
* Adds 404 error codes to file endpoints in core-api
* Adds unit tests to pick up these error codes
* Converts unit tests to using `http.HTTPStatus` instead of hard-coded error numbers
* Edits local E2E tests so they run and don't destroy your `.env`

## Guidance to review

* Check you're happy with the RFC-lite approach
* Ensure you're happy with my specific implementation of the error codes -- have I included the right fields? Is my copy what you'd expect? Would you be happy extending these objects?
* Check you're happy with my changes to the E2E tests. Are these variables okay in `pyproject.toml` rather than `.env.integration`?

## Relevant links

* [RFC 9457 API error specification](https://www.rfc-editor.org/rfc/rfc9457.html)
* [Additional Responses in OpenAPI](https://fastapi.tiangolo.com/advanced/additional-responses/#additional-responses-in-openapi), which shows how to add and document error codes

## Things to check

- [x] I have added any new ENV vars in all deployed environments
- [x] I have tested any code added or changed
- [x] I have run integration tests

### CODE REVIEW

File: core_api/src/routes/file.py [hunk 1]
```diff
@@ -115,10 +117,29 @@ def get_file(file_uuid: UUID, user_uuid: Annotated[UUID, Depends(get_user_uuid)]
 
     Returns:
         File: The file
+
+    Raises:
+        404: If the file isn't found, or the creator and requester don't match
     """
-    file = storage_handler.read_item(file_uuid, model_type="File")
+    not_found = JSONResponse(
```
Comment (brunns) [inline core_api/src/routes/file.py:124]: Nitpick - you could move the creation of this object into a function - `def file_not_found_response(file_uuid: UUID) -> JSONResponse` or the like. Then, don't call this function yet...

File: core_api/src/routes/file.py [hunk 2]
```diff
@@ -115,10 +117,29 @@ def get_file(file_uuid: UUID, user_uuid: Annotated[UUID, Depends(get_user_uuid)]
...
+                "parameter": "file_uuid",
+                "detail": f"File {file_uuid} not found",
+            },
+        },
+    )
+
+    try:
+        file = storage_handler.read_item(file_uuid, model_type="File")
+    except NotFoundError:
+        return not_found
```
Comment (brunns) [inline core_api/src/routes/file.py:138]: `return file_not_found_response(file_uuid)`

File: core_api/src/routes/file.py [hunk 3]
```diff
@@ -115,10 +117,29 @@ def get_file(file_uuid: UUID, user_uuid: Annotated[UUID, Depends(get_user_uuid)]
...
+
     if file.creator_user_uuid != user_uuid:
-        raise HTTPException(status_code=404, detail=f"File {file_uuid} not found")
+        return not_found
```
Comment (brunns) [inline core_api/src/routes/file.py:141]: `return file_not_found_response(file_uuid)`

File: core_api/tests/routes/test_file.py [hunk 4]
```diff
@@ -47,6 +47,16 @@ def test_get_file(app_client, stored_file, headers):
...
 
 
+def test_get_missing_file(app_client, headers):
+    """
+    Given a nonexistent file
+    When I GET it from /file/uuid
+    I Expect to receive a 404 error
+    """
+    response = app_client.get("/file/ffffffff-ffff-ffff-ffff-ffffffffffff", headers=headers)
+    assert response.status_code == 404
```
Comment (brunns) [inline core_api/tests/routes/test_file.py:57]: Add
```
from http import HTTPStatus
```

Then assert:

```
assert response.status_code == HTTPStatus.NOT_FOUND
```

**Review by brunns — APPROVED** [review]
Nitpicks.

File: Makefile [hunk 5]
```diff
@@ -39,15 +39,11 @@ test-django:
 
 test-integration:
 	docker compose down
-	cp .env .env.backup
-	cp .env.integration .env
```
Comment (brunns) [inline Makefile:43]: Does it work locally without this?

**Review by brunns — APPROVED** [review]
You might want to check the Makefile check with @gecBurton. Aside from that, LGTM.

**Review by gecBurton — APPROVED** [review]
lovely stuff!

File: core_api/src/routes/file.py [hunk 6]
```diff
@@ -115,14 +138,27 @@ def get_file(file_uuid: UUID, user_uuid: Annotated[UUID, Depends(get_user_uuid)]
...
-        raise HTTPException(status_code=404, detail=f"File {file_uuid} not found")
+        return file_not_found_response(file_uuid=file_uuid)
+
     return file
 
 
-@file_app.delete("/{file_uuid}", response_model=File, tags=["file"])
+@file_app.delete(
+    "/{file_uuid}",
+    response_model=File,
```
Comment (gecBurton) [inline core_api/src/routes/file.py:158]: you dont actually need this here of you include it as an annotation below?

File: core_api/tests/routes/test_file.py [hunk 7]
```diff
@@ -59,7 +70,7 @@ def test_delete_file(s3_client, app_client, elasticsearch_storage_handler, chunk
     assert elasticsearch_storage_handler.get_file_chunks(chunked_file.uuid, chunked_file.creator_user_uuid)
 
     response = app_client.delete(f"/file/{chunked_file.uuid}", headers=headers)
-    assert response.status_code == 200
+    assert response.status_code == HTTPStatus.OK
```
Comment (gecBurton) [inline core_api/tests/routes/test_file.py:73]: 👍

File: redbox/models/errors.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
+from pydantic import AnyUrl, BaseModel, Field
+
+
+class APIErrorDetail(BaseModel):
+    """A single API error loosely defined to the RFC 9457 format."""
```
Comment (gecBurton) [inline redbox/models/errors.py:5]: ❤️

**Pull request merged**